### DISCUSSION TITLE
Interactive message

### DISCUSSION THREAD

**Discussion by alessandro-p-farias** [tulir/whatsmeow#711] [WhatsApp protocol Q&A]
Hi, I'm trying to send an interactive message, but the message only shows header, body e footer. The button and sections to choose is not displayed.
I think that is a problem related to the parameter ButtonParamsJSON inside nativeFlowMessage, but i can't find any help or examples of what I shoul send in this parameter.
Example of the json converted value I'm sending:
```
{
  "title": "Main Title",
  "ButtomText": "Choose",
  "sections": [
    {
      "title": "Section 1",
      "rows": [
        {
          "title": "Row 1",
          "description": "Description 1",
          "id": "1-abc123"
        },
        {
          "title": "Row 2",
          "description": "Description 2",
          "id": "2-abc123"
        }
      ]
    }
}
```

Code to send message:
```
msg := &waProto.Message{
					MessageContextInfo: &waProto.MessageContextInfo{
						DeviceListMetadataVersion: proto.Int32(3),
					},
					InteractiveMessage: &waProto.InteractiveMessage{
						InteractiveMessage: &waProto.InteractiveMessage_NativeFlowMessage_{
							NativeFlowMessage: &waProto.InteractiveMessage_NativeFlowMessage{
								Buttons: []*waProto.InteractiveMessage_NativeFlowMessage_NativeFlowButton{
									{
										Name: proto.String("button_1"),
										ButtonParamsJSON: proto.String(string(jsonButton)),
									},
								},
								MessageParamsJSON: proto.String(""),
							},
						},
						Header: &waProto.InteractiveMessage_Header{
							Title: proto.String("test header"),
							HasMediaAttachment: proto.Bool(false),
						},
						Body: &waProto.InteractiveMessage_Body{
							Text: proto.String("test body"),
						},
						Footer: &waProto.InteractiveMessage_Footer{
							Text: proto.String("test footer"),
						},
					},
				}
				result, err := client.SendMessage(context.Background(), v.Info.Chat, msg)
				fmt.Println(msg, result, err)
```

message I'm receiving
![image](https://github.com/user-attachments/assets/a91287de-c278-423d-b729-fd8f29afbedb)

Can anyone help me?

**Comment by masewo** [comment]
Button messages are not supported.

**Comment by alessandro-p-farias** [comment]
Ok, Thanks. But I'm not trying to use button message, I was trying to send an "option list" message with single select options like the example below...

![image](https://github.com/user-attachments/assets/312370ce-3b04-464e-b368-c3b534c602b9)

Is possible to send a message like this with whatsmeow? If it is, can you point me to the right direction?

**Reply by Adamkaram** [reply]
did u figure out?

**Comment by SamuelGaona** [comment]
I think that is possible only with the official API.
